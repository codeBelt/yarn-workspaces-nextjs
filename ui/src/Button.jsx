import React from 'react';
import { kebabCase } from '@codebelt/util/src/stringUtil';

export const Button = props => {
  const kebabCaseName = kebabCase(props.name);

  return <button onClick={() => alert('Robert is Cool')}>{kebabCaseName}</button>;
};
