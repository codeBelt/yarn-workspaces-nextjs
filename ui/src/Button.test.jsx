import React from 'react';
import renderer from 'react-test-renderer';
import {Button} from "./Button";

jest.spyOn(window, 'alert').mockImplementation(() => {});

describe('Button', () => {
  test('Snapshot and onClick', () => {
    const component = renderer.create(
        <Button name="Press Me" />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onClick();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
