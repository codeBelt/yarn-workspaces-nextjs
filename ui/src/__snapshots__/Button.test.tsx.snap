// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot and onClick 1`] = `
<button
  onClick={[Function]}
>
  press-me
</button>
`;

exports[`Button Snapshot and onClick 2`] = `
<button
  onClick={[Function]}
>
  press-me
</button>
`;
